tests/user-envs.sh

```diff
@@ -179,3 +180,15 @@ nix-env --set $outPath10
 [ "$(nix-store -q --resolve $profiles/test)" = $outPath10 ]
 nix-env --set $drvPath10
 [ "$(nix-store -q --resolve $profiles/test)" = $outPath10 ]
+
+# This tests behavior of the daemon, and as such only needs to run
+# when the daemon is the self version
+if [[ -n "${NIX_CLIENT_PACKAGE:-}" && -n "${NIX_DAEMON_PACKAGE:-}"
+    && "$NIX_CLIENT_PACKAGE" == "$NIX_DAEMON_PACKAGE" ]]; then
```

Comment:
Let's see if this works. This should only make the test run when client version matches daemon version.

Reply:
It does :)